File: VB/WindowsFormsApplication1/BarButtonTabPage.vb
Imports Microsoft.VisualBasic
Imports System.Drawing
Imports DevExpress.XtraBars
Imports DevExpress.XtraReports.Design
Imports DevExpress.XtraReports.UserDesigner

Namespace Common
	Friend Class BarButtonTabPage
		Inherits XtraTabControl.TabPage
		Private ReadOnly barButtonItem As BarButtonItem
		Private wpfReportPreviewUserControl As WindowsFormsApplication1.WPF.WpfReportPreviewUserControl
		Private Caption As String
		Private reportCommand As ReportCommand

		Public Sub New(ByVal wpfReportPreviewUserControl As WindowsFormsApplication1.WPF.WpfReportPreviewUserControl, ByVal Caption As String, ByVal button As BarButtonItem, ByVal reportCommand As ReportCommand)
			MyBase.New(wpfReportPreviewUserControl, Caption, reportCommand)
			Me.wpfReportPreviewUserControl = wpfReportPreviewUserControl
			Me.Caption = Caption
			Me.reportCommand = reportCommand
			Me.barButtonItem = button
		End Sub
		Public Overrides Property Selected() As Boolean
			Get
				Return MyBase.Selected
			End Get
			Set(ByVal value As Boolean)
				If MyBase.Selected <> value Then
					barButtonItem.Down = value
				End If
				MyBase.Selected = value
			End Set
		End Property
		Public Overrides Property Visible() As Boolean
			Get
				Return MyBase.Visible
			End Get
			Set(ByVal value As Boolean)
				MyBase.Visible = value
				barButtonItem.Visibility = If(value, BarItemVisibility.Always, BarItemVisibility.Never)
			End Set
		End Property
		Public Overrides Property Image() As Image
			Get
				Return barButtonItem.Glyph
			End Get
			Set(ByVal value As Image)
				barButtonItem.Glyph = value
			End Set
		End Property
	End Class
End Namespace

File: CS/WindowsFormsApplication1/BarButtonTabPage.cs
using System.Drawing;
using DevExpress.XtraBars;
using DevExpress.XtraReports.Design;
using DevExpress.XtraReports.UserDesigner;

namespace Common {
    class BarButtonTabPage : XtraTabControl.TabPage {
        readonly BarButtonItem barButtonItem;
        private WindowsFormsApplication1.WPF.WpfReportPreviewUserControl wpfReportPreviewUserControl;
        private string Caption;
        private ReportCommand reportCommand;

        public BarButtonTabPage(WindowsFormsApplication1.WPF.WpfReportPreviewUserControl wpfReportPreviewUserControl, string Caption, BarButtonItem button, ReportCommand reportCommand)
            : base(wpfReportPreviewUserControl, Caption, reportCommand) {
            this.wpfReportPreviewUserControl = wpfReportPreviewUserControl;
            this.Caption = Caption;
            this.reportCommand = reportCommand;
            this.barButtonItem = button;
        }
        public override bool Selected {
            get { return base.Selected; }
            set {
                if (base.Selected != value)
                    barButtonItem.Down = value;
                base.Selected = value;
            }
        }
        public override bool Visible {
            get {
                return base.Visible;
            }
            set {
                base.Visible = value;
                barButtonItem.Visibility = value ? BarItemVisibility.Always : BarItemVisibility.Never;
            }
        }
        public override Image Image {
            get { return barButtonItem.Glyph; }
            set { barButtonItem.Glyph = value; }
        }
    }
}